plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/linked-product-tab-product-block-editor.spec.js

```diff
@@ -86,15 +86,19 @@ test.describe( 'General tab', { tag: '@gutenberg' }, () => {
 
 				// Include in category
 				await clickOnTab( 'Organization', page );
-				const waitForCategoriesResponse = page.waitForResponse(
-					( response ) =>
-						response
-							.url()
-							.includes( '/wp-json/wp/v2/product_cat' ) &&
-						response.status() === 200
-				);
 				await page.getByLabel( 'Categories' ).click();
-				await waitForCategoriesResponse;
+				await page
+					.getByLabel( 'Categories' )
+					.type( categoryName, { delay: 100 } );
+				await page
+					.getByRole( 'option', { name: categoryName } )
+					.locator( 'label' )
+					.waitFor();
+				await expect(
+					page
+						.getByRole( 'option', { name: categoryName } )
+						.locator( 'label' )
+				).toHaveText( categoryName );
```

Comment:
I'm don't think removing `waitForCategoriesResponse` is a good idea. This is a very dynamic area and I remember having a lot of trouble stabilizing it and waitForCategoriesResponse fixed it. I see no hard in leaving it. And I'm very sure it's not a cause of flakiness, so why remove it?

Reply:
> I added action to click and type the category name just like a normal end user would type, which adds some time and shows the category in the end.

Waiting for 100ms does nothing to guarantee the categories are loaded.
The `waitForCategoriesResponse` is there to make sure the categories are fetched before typing. If they are not, the test will fail.

Reply:
I understand the reasoning behind keeping `waitForCategoriesResponse` — it worked in the past to stabilize the test, and I can see how that was necessary given the dynamic nature of this area. However, I encountered a specific issue where the test failed due to the timing of the response, and I believe waiting for the response alone wasn’t enough to guarantee the categories were loaded in time.

The test was failing intermittently while waiting for the response from waitForCategoriesResponse, and this was the root cause of the issue. The delay of 100ms between keystrokes I added to simulate typing was working well in stabilizing the input, but it wasn’t enough to handle the timing of the response in some cases.

As a result, I added a more explicit wait to ensure that the category option was not only typed into the field but also loaded and visible before interacting with it. The combination of waiting for the option’s label to appear and asserting the text seemed to be an effective way to prevent flaky failures, and I’ve executed it locally several times with consistent success.

I think the solution I implemented is more robust because it:

1. Ensures categories are fully loaded before typing by waiting for the category option’s label to appear.
2. Avoids unnecessary waits — it doesn’t just blindly wait for the response but explicitly ensures the option is available before proceeding.

While `waitForCategoriesResponse` helped us in the past, I believe this additional wait for the category option’s label ensures better synchronization, reducing the chances of race conditions and preventing test failures.

I’m happy to discuss further if needed and hear from others as well.

Reply:
> * Ensures categories are fully loaded before typing by waiting for the category option’s label to appear.

I'm sorry but this is not how it works. If you start typing before the categories are fetched it will fail. Typing will display the list of categories, and if the fetch finishes after the typing starts the list is not updated and the category will not be found.

> * Avoids unnecessary waits — it doesn’t just blindly wait for the response but explicitly ensures the option is available before proceeding.

This is not how waiting for the response works. There's nothing blind about it.

Reply:
Oh, this test `linked-product-tab-product-block-editor.spec.js` is not even reported as flaky. Why would you "fix" it? 
> The test was failing intermittently while waiting for the response from waitForCategoriesResponse

Reply:
@adimoldovan , I have nothing against your implementation. I found it flaky and [here for example](https://github.com/woocommerce/woocommerce/actions/runs/12068850803/job/33655488067) it failed twice in a row. However, since we are dealing with such a low flakiness score and it is working well for the last many runs, let's just keep it as is. We could even combine both implementations if you'd like.

Please give it another look and let me know if I can merge it.

Reply:
@veljkho 
1. That is a PR in progress. 
2. That is not a flaky occurrence. In that run the test failed - it is broken. Coincidently that is a PR I am working on and that failure was expected in that case.

Do you have a flakiness report on this test?